std/process.d

```diff
@@ -682,6 +685,12 @@ private Pid spawnProcessImpl(scope const(char[])[] args,
     }
 }
 
+private struct SpawnContext
```

Comment:
There is no need to add a struct type if you're going to add a new argument. The reason I suggested using a struct was because a newly added argument would be unused in the POSIX version. The POSIX version doesn't need this informaiton, so there is no need to change its signature at all; but, if you're going to add a new argument, there is no need to introduce a struct.

Reply:
If the signatures between windows and posix don't match then we'll need to have 2 different calls to these functions inside version statements.  Is that what you want?

Reply:
This is not true. They already don't match in `master`.

Reply:
didn't compile when they had different signatures, I've pushed the change with the versions.  Not sure what you want to do here.

Reply:
> didn't compile when they had different signatures

Please take a moment to have a closer look at the signatures of the Windows and POSIX versions of `spawnProcessImpl`, as they are in master right now (before your changes).

>  Not sure what you want to do here.

Would actually putting in a little effort be too much to ask? :(

Reply:
I'm reading alot of hostility coming from you right now and I have no idea why?  Why are you being so aggressive and insulting me?

I'm looking at master and they have the same signatures.

https://github.com/dlang/phobos/blob/master/std/process.d#L355
https://github.com/dlang/phobos/blob/master/std/process.d#L695

Reply:
> I'm looking at master and they have the same signatures.

Please, look a little closer...

One of them takes an array, the other takes a string.

My suggestion was to replace the array in the Windows version with a `struct`. Then, no changes to the POSIX signature or additional `version` blocks would be required.

> I'm reading alot of hostility coming from you right now and I have no idea why? Why are you being so aggressive and insulting me?

Sorry, I feel like you are abusing my time by not spending a little more of your time to research this issue and submit a good contribution.

If you like, let's close this PR and I will submit one which addresses this issue in a manner I consider to be optimal.

BTW you still need to file an issue in Bugzilla for the changelog.

Reply:
I don't care much how this is implemented.  I can see that you care very much, so feel free to let me know what you want.  There's no need for saying things like "would actually putting in a little effort be too much to ask".  I'm willing to do what you want, and I don't think I've given you any reason to think otherwise or any reason to think that I'm not willing to put in work.  I've already completely re-written this thing 3 times in the last 40 minutes based on your suggestions but I can't read your mind to know what you want.

Reply:
I don't see why you think reading my mind is required. I wrote:

> I suggest changing the first argument of the Windows version of `spawnProcessImpl` to a `struct` type or tuple

I don't see how that's unclear.